### DISCUSSION TITLE
Error creating UDS listener: listen unix /var/run/datadog/apm.socket: bind: read-only file system

### DISCUSSION THREAD

**Discussion by diegovss** [DataDog/datadog-agent#29300] [Q&A]
Since the update to the latest version 7.57.0, my datadog agent keep loggin the following line:
`UTC | TRACE | CRITICAL | (pkg/trace/api/api.go:712 in func2) | Error creating UDS listener: listen unix /var/run/datadog/apm.socket: bind: read-only file system`

Also, the metrics of APM disapeared. Do you guys know how to fix this?

I saw this in the enhancement notes for the version 7.57.0:

- APM: Add default UDS listeners for traces (trace-agent) and dogstatsd (core-agent) on /var/run/datadog/apm.socket and /var/run/datadog/dsd.socket, respectively. These are used in the Single Step APM Instrumentation, improving the onboarding experience and minimizing the agent configuration.

**Comment by KebiLab** [comment]
How to fix:
Disable the listeners by adding these environment variables to your Agent:
yaml

DD_APM_RECEIVER_SOCKET: ""
DD_DOGSTATSD_SOCKET: ""

Alternatively, if you want to use UDS, mount a writable volume to /var/run/datadog/.
